apollo-ios/Sources/Apollo/GraphQLResult.swift

```diff
@@ -151,7 +151,12 @@ fileprivate extension DataDict {
     }
 
     let mergedData = try pathDataDict._data.merging(newDataDict._data) { current, new in
-      throw Error.cannotOverwriteData(current, new)
+      // TODO: This is a quick fix for __typename being returned with fragments, needs a rethink!
```

Comment:
To throw here seems the correct behaviour because there should never be a case of overwriting response data; incremental data is only ever additive. The exception to this rule is `__typename` which is being returned for fragments on the same type case and already exists in the base data set.